File: MsgTools/struct/java/AllStruct.java
package @packages@;

import com.fantingame.game.msgbody.common.io.iface.IXInputStream;
import com.fantingame.game.msgbody.common.io.iface.IXOutStream;
import com.fantingame.game.msgbody.common.model.ICodeAble;
import java.io.IOException;

@imports@
/**@note@**/
public class @className@ implements ICodeAble {

	@param@
	
	@Override
	public void encode(IXOutStream outputStream) throws IOException {
@encode@
	}

	@Override
	public void decode(IXInputStream inputStream) throws IOException {
@decode@
	}
	
	@getterAndSetter@
	
	@otherMethod@
}

File: MsgTools/struct/lua/AllStruct.lua
@imports@

@className@ = {}

--@note@
function @className@:New(o)
    o = o or {}
    setmetatable(o,self)
    self.__index = self
	self:init()
    return o
end

function @className@:init()
	@param@
	self.actName = "@actName@"
end

function @className@:getActName()
	return self.actName
end

@getterAndSetter@


@otherMethod@

function @className@:encode(outputStream)
@encode@
end

function @className@:decode(inputStream)
	    local body = {}
@decode@
	   return body
end